services/reuse/reuse-compliance.service.js

```diff
@@ -0,0 +1,59 @@
+'use strict'
+const Joi = require('joi')
+const { BaseJsonService } = require('..')
+const { renderReuseBadge } = require('./reuse-compliance-helper')
+
+const responseSchema = Joi.object({
+  badge: Joi.string(),
+  hash: Joi.string(),
+  last_access: Joi.date(),
+  lint_code: Joi.number(),
+  lint_output: Joi.string(),
+  status: Joi.string().required(),
+  url: Joi.string().required(),
+}).required()
+
+module.exports = class Reuse extends BaseJsonService {
+  static category = 'license'
+
+  static route = {
+    base: 'reuse/compliance',
+    pattern: ':gitserver/:user/:repository',
+  }
+
+  static examples = [
+    {
+      title: 'REUSE Compliance',
+      namedParams: {
+        gitserver: 'github',
+        user: 'fsfe',
+        repository: 'reuse-tool',
+      },
+      staticPreview: this.render({ status: 'compliant' }),
+      keywords: ['license'],
+    },
+  ]
+
+  static defaultBadgeData = {
+    label: 'REUSE',
+  }
+
+  static render({ status }) {
+    return renderReuseBadge({ status })
+  }
+
+  async fetch({ gitserver, user, repository }) {
+    return await this._requestJson({
+      schema: responseSchema,
+      url: `https://api.reuse.software/status/${gitserver}.com/${user}/${repository}`,
```

Comment:
What's the deal with this `gitserver` param? Is there a list of compatible services, like this
https://github.com/badges/shields/blob/66a73b9f987685fb0a500bb0607d568d57deefb8/services/coveralls/coveralls.service.js#L22
or can any git remote be used? If so, is .com really a requirement?

Reply:
The API provided by `REUSE` supports all repositories that use `git`.  And in the API format, the `.com` is necessary.

Here is the example to try: 
https://api.reuse.software/status/github.com/fsfe/reuse-tool 

Reference: https://api.reuse.software/register

Reply:
ok, but if my git remote is not on a `.com` domain (`.org`, `.net` etc e.g: https://gitlab.gnome.org ) then this isn't going to work.

There's another case that may give us a problem which is where your git remote is not hosted at the root of the domain. I don't have an example of this off the top of my head to see how REUSE handles it, but I'll try and find one tomorrow to check (if anyone happens to know one please chime in - this may have come up before). We might have to convert this to a URL param as we have a hard limitation from path-to-regexp that we can't have a capturing group capable of containing a slash unless it is the last capturing group in the route.

Reply:
I had a bit more of a think about this and I think the way that actually makes most sense if any git remote can be used (including something like https://git.eclipse.org/c/4diac/org.eclipse.4diac.ide.git/ where the clone URLs are under a subdir) is to have a single `:remote+` param (which can then contain an arbitrary number of slashes). So the code is something like:

```js
  static route = {
    base: 'reuse/compliance',
    pattern: ':remote+',
  }

  async fetch({ remote }) {
    return await this._requestJson({
      url: `https://api.reuse.software/status/${remote}`,
    })
  }
```

and then the badge URLs will be like:

* `/reuse/compliance/github.com/fsfereuse-tool`
* `/reuse/compliance/gitlab.gnome.org/GNOME/gnome-shell`
* `/reuse/compliance/git.eclipse.org/gitroot/4diac/org.eclipse.4diac.ide`

We can add little HTML fragment explaining the URL format in the `documentation` key of an `example` object.

Reply:
This is a awesome idea!